src/compute/src/render/continual_task.rs

```diff
@@ -189,12 +190,15 @@ impl<G: Scope<Timestamp = Timestamp>> ContinualTaskCtx<G> {
         Some(inserts_source_fn)
     }
 
-    pub fn input_times(&self) -> Option<Collection<G, (), Diff>> {
-        let (Some(name), Some(first)) = (self.name.as_ref(), self.ct_times.first()) else {
+    pub fn input_times(&self, scope: &G) -> Option<Collection<G, (), Diff>> {
+        if !self.is_ct_dataflow() {
             return None;
         };
+        let Some(name) = self.name.as_ref() else {
+            return Some(Collection::empty(scope));
+        };
```

Comment:
Isn't this dead code? Looking at the definitions of `new` and `is_ct_dataflow`, we should always have a name if we have a CT dataflow (there is even an assert that checks this).

If it's not dead code I'd appreciate a comment saying why! Also why we return an empty collection instead of `None` in that case.

Reply:
Ah good eyes, this is just an artifact of how this branch evolved when I was iterating on it. Simplified the logic and added a comment about what it means for ct_inputs to be empty.